From foo@baz Tue Apr  9 12:12:43 2002
To: Greg KH <greg@kroah.com>
Date: 17 Feb 2005 15:06:16 -08:00
From: ak@muc.de
Subject: [PATCH] PCI: allow x86_64 to do pci express

On Mon, Feb 14, 2005 at 10:47:01AM +0100, Piotr Kaczuba wrote:
> On Mon, Feb 14, 2005 at 10:18:43AM +0100, Andi Kleen wrote:
> > Piotr Kaczuba <pepe@attika.ath.cx> writes:
> > > Is there a reason why "PCI access mode" config option isn't available for
> > > x86_64? Due to this, PCIE config options aren't available either.
> >
> > There is no 64bit PCI BIOS, so access is always direct.
> >
> > I assume you mean mmconfig access with "PCIE config options", that is
> > a separate config option and available.
>
> I mean the PCIEPORTBUS option which depends on PCI_GOMMCONFIG or
> PCI_GOANY. I assume that due to PCI_MMCONFIG / PCI_GOMMCONFIG mismatch
> it's not available on x86_64.

Ok, that's a bug in PCIEPORTBUS.  Best is probably to
completely remove the dependency, it doesn't make much sense
(the code has to handle the case of mmconfig not being available at
runtime anyways)

Remove bogus dependency in PCI Express root driver.

Signed-off-by: Andi Kleen <ak@suse.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


diff -Nru a/drivers/pci/pcie/Kconfig b/drivers/pci/pcie/Kconfig
--- a/drivers/pci/pcie/Kconfig	2005-03-02 23:04:39 -08:00
+++ b/drivers/pci/pcie/Kconfig	2005-03-02 23:04:39 -08:00
@@ -3,7 +3,6 @@
 #
 config PCIEPORTBUS
 	bool "PCI Express support"
-	depends on PCI_GOMMCONFIG || PCI_GOANY
 	default n
 
 	---help---
